src/psyclone/psyir/nodes/commentable_mixin.py

```diff
@@ -0,0 +1,86 @@
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author S. Siso, STFC Daresbury Lab
+# -----------------------------------------------------------------------------
+
+''' This module contains the CommentableMixin implementation.'''
+
+import six
+
+
+class CommentableMixin:
+    '''
+    Mixin that adds the Commentable trait into a PSyIR node. It provides
+    two attributes that store preceding and inline comments and their
+    respective property getters and setters.
+    '''
+
+    _preceding_comment = ""
+    _inline_comment = ""
+
+    @property
+    def preceding_comment(self):
+        '''
+        :returns: string comment preceding this statement.
+        :rtype: str
+        '''
+        return self._preceding_comment
+
+    @preceding_comment.setter
+    def preceding_comment(self, comment):
+        '''
+        :param str comment: string comment preceding this statement.
+        '''
+        if not isinstance(comment, six.string_types):
+            raise TypeError("The preceding_comment must be a string but"
+                            " found '{0}'.".format(type(comment).__name__))
+        self._preceding_comment = comment
+
+    @property
+    def inline_comment(self):
+        '''
+        :returns: string inline comment this statement.
+        :rtype: str
+        '''
+        return self._inline_comment
+
+    @inline_comment.setter
+    def inline_comment(self, comment):
+        '''
+        :param str comment: string comment preceding this statement.
+        '''
+        if not isinstance(comment, six.string_types):
+            raise TypeError("The inline_comment must be a string but"
+                            " found '{0}'.".format(type(comment).__name__))
+        self._inline_comment = comment
```

Comment:
Please could you add an `__all__`  at the bottom of this file.

Reply:
Added `__all__`